### PR TITLE
Support tree-shaking and split code into entrypoints

### PR DESCRIPTION
**Pull request by soerenbf**
## Purpose

Make it possible to include only the functionality you need from the SDK for your concordium web3 application.

## Changes

- Split code into several entrypoints (to support commonJS not loading everything)
- Publish web-sdk as ESM
- Split WASM module into
  - functionality commonly needed by dApps
  - functionality needed by wallets

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Review by limemloh — APPROVED** [review]
Looks good

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# docs
```
[the remark this answers is not in the document]
Comment (soerenbf) [reply docs/README.md:1]: Haha, true.

File: examples/ccd-js-gen/wCCD/generate.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import * as Gen from '@concordium/ccd-js-gen';
 import * as Path from 'node:path';
 import * as Url from 'node:url';
 import meow from 'meow';
```
Comment (limemloh) [inline examples/ccd-js-gen/wCCD/generate.ts:6]: This example seems to fail building now

File: packages/common/src/GenericContract.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { Buffer } from 'buffer/';
 import { stringify } from 'json-bigint';
```
Comment (limemloh) [inline packages/common/src/GenericContract.ts:2]: I could be overlooking something, but I don't see how `Contract` in this file is public anymore. The client generator relies on it being public

Comment (soerenbf) [reply packages/common/src/GenericContract.ts:2]: Must have forgotten to export it. Will add this.

**Pull request merged**